pkg/webhook/cluster/nutanix_validator.go

```diff
@@ -89,9 +90,113 @@ func (a *nutanixValidator) validate(
 		}
 	}
 
+	if err := validateFailureDomainRelatedConfig(clusterConfig, cluster); err != nil {
+		return admission.Denied(err.Error())
+	}
+
 	return admission.Allowed("")
 }
 
+// validateFailureDomainRelatedConfig validates the failure domain related configuration in cluster topology.
+func validateFailureDomainRelatedConfig(
+	clusterConfig *variables.ClusterConfigSpec,
+	cluster *clusterv1.Cluster,
+) error {
+	fldErrs := field.ErrorList{}
+	fldPath := field.NewPath("spec", "topology")
+	// Validate that either failureDomains is set, or cluster and subnets are set with machineDetails, for control plane.
+	if clusterConfig.ControlPlane != nil &&
+		clusterConfig.ControlPlane.Nutanix != nil &&
+		len(clusterConfig.ControlPlane.Nutanix.FailureDomains) == 0 {
+		machineDetails := clusterConfig.ControlPlane.Nutanix.MachineDetails
+		if machineDetails.Cluster == nil || (!machineDetails.Cluster.IsName() && !machineDetails.Cluster.IsUUID()) {
```

Comment:
do IsName and IsUUID give mutually exclusive results? is there a way to simplify this if statement?